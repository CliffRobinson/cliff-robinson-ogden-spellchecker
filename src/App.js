import {useState} from 'react'

import './App.css';
import {UserInputBox} from './components/UserInputBox'
import {OutputBox} from './components/OutputBox'
import {checkSentence} from './checkerFunctions'

//CFR Comment 1: The app as described has no need to access a backend, 
//I'd like to have a browser GUI if possible, and React is my framework of choice,
//so create-react-app is a great place to start. 
//I like to practise TDD where possible, so let's get Enzyme set up for snapshots next. 

//CFR Comment 3: This is a first pass at how I generally want the app to behave - one
// area for the user to input text, and another that displays the text with markup to
// show which words are mispelled. Both inputs are going to be stateless components, with
// app acting as a container and store for the user input string. For now the next step is 
// going to be breaking out the user input box into a component.
function App() {

  const [userInputString, setUserInputString] = useState("initial string")

  return (
    <div className="App">
      <h3>Ogden's Simplified Spellchecker!</h3>
      {<UserInputBox 
        userInputString={userInputString}
        setUserInputString={setUserInputString}
      />}
      {<OutputBox wordArray={checkSentence(userInputString)}/>}
    </div>
  );
}

export default App;
