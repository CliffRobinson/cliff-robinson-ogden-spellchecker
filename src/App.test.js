import {shallow} from 'enzyme'

import App from './App';

test('app render', () => {
  const tree = shallow(<App />)
  expect(tree).toMatchSnapshot()
})