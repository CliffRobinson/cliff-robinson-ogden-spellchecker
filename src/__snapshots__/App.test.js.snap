// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app render 1`] = `
<div
  className="App"
>
  <h3>
    Ogden's Simplified Spellchecker!
  </h3>
  <UserInputBox
    setUserInputString={[Function]}
    userInputString="it is hard to make good sentences with this dictionary"
  />
  <OutputBox
    wordArray={
      Array [
        Object {
          "spelling": undefined,
          "word": "it",
        },
        Object {
          "spelling": undefined,
          "word": "is",
        },
        Object {
          "spelling": true,
          "word": "hard",
        },
        Object {
          "spelling": true,
          "word": "to",
        },
        Object {
          "spelling": true,
          "word": "make",
        },
        Object {
          "spelling": true,
          "word": "good",
        },
        Object {
          "spelling": undefined,
          "word": "sentences",
        },
        Object {
          "spelling": true,
          "word": "with",
        },
        Object {
          "spelling": true,
          "word": "this",
        },
        Object {
          "spelling": undefined,
          "word": "dictionary",
        },
      ]
    }
  />
</div>
`;
